### ISSUE TITLE
Can't build qtum mainnet-ignition-v0.18.1

### ISSUE THREAD

**Issue by gituser** [qtumproject/qtum#767]
Can't build the latest version of qtum `mainnet-ignition-v0.18.1` on Ubuntu 16.04:

```
In file included from ./cpp-ethereum/libethereum/CommonNet.h:28:0,
                 from ./cpp-ethereum/libethereum/ClientBase.h:30,
                 from ./cpp-ethereum/libethereum/Client.h:27,
                 from ./cpp-ethereum/libethashseal/Ethash.h:25,
                 from ./validation.h:44,
                 from blockencodings.cpp:14:
./cpp-ethereum/libp2p/Common.h: In member function 'dev::p2p::Node::operator bool() const':
./cpp-ethereum/libp2p/Common.h:259:52: error: cannot convert 'const NodeID {aka const dev::FixedHash<64u>}' to 'bool' in initialization
     explicit operator bool() const { return bool{id}; }
                                                    ^
Makefile:11702: recipe for target 'libbitcoin_server_a-blockencodings.o' failed
make[2]: *** [libbitcoin_server_a-blockencodings.o] Error 1
make[2]: Leaving directory '/home/build/qtum/source/src'
Makefile:18312: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/build/qtum/source/src'
Makefile:787: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

Previous version of qtum `mainnet-ignition-v0.18.0` builds just fine on the same setup.

Anything changed lately in the build requirements?

Thanks.

**Comment by gituser** [comment]
Ping @qtum-neil

**Comment by qtum-neil** [comment]
Seems like you are using an old compiler version, can you please post your system specs and compiler version?

**Comment by gituser** [comment]
@qtum-neil 

yes, I'm using stock gcc shipped with Ubuntu 16.04:

```
gcc -v                   
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/5/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 5.4.0-6ubuntu1~16.04.11' --with-bugurl=file:///usr/share/doc/gcc-5/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-5 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-5-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-5-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-5-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.11) 
```

**Comment by qtum-neil** [comment]
You need to either use a newer gcc version (7.4.0 is working), or patch src/cpp-ethereum/libp2p/Common.h file like this:

https://github.com/ethereum/aleth/commit/aa2a03551ff601578e07cd85b984c15da06c3958
